exercise/src/main/java/co/edu/etitc/sistemas/tecnologo/programacion4/controllers/BibliotecaController.java

```diff
@@ -0,0 +1,63 @@
+package co.edu.etitc.sistemas.tecnologo.programacion4.controllers;
+
+import co.edu.etitc.sistemas.tecnologo.programacion4.*;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.Collection;
+
+@RestController
+@RequestMapping("/api/recursos")
+public class BibliotecaController {
+
+    private final ServicioBiblioteca servicio;
+    private final LibroRepositorio libroRepositorio;
+    private final PeriodicoRepositorio periodicoRepositorio;
+    private final ComputadorRepositorio computadorRepositorio;
+
+    public BibliotecaController(ServicioBiblioteca servicio,
+                              LibroRepositorio libroRepositorio,
+                              PeriodicoRepositorio periodicoRepositorio,
+                              ComputadorRepositorio computadorRepositorio) {
+        this.servicio = servicio;
+        this.libroRepositorio = libroRepositorio;
+        this.periodicoRepositorio = periodicoRepositorio;
+        this.computadorRepositorio = computadorRepositorio;
+    }
+
+    @GetMapping
+    public Collection<Recurso> obtenerTodos() {
+        return servicio.obtenerTodos();
+    }
+
+    @PostMapping("/libros")
+    public ResponseEntity<Libro> agregarLibro(@RequestBody Libro libro) {
+        Libro libroGuardado = libroRepositorio.save(libro);
+        return ResponseEntity.ok(libroGuardado);
+    }
+
+    @PostMapping("/periodicos")
+    public ResponseEntity<Periodico> agregarPeriodico(@RequestBody Periodico periodico) {
+        Periodico periodicoGuardado = periodicoRepositorio.save(periodico);
+        return ResponseEntity.ok(periodicoGuardado);
+    }
+
+    @PostMapping("/computadores")
+    public ResponseEntity<Computador> agregarComputador(@RequestBody Computador computador) {
+        Computador computadorGuardado = computadorRepositorio.save(computador);
+        return ResponseEntity.ok(computadorGuardado);
+    }
+
+        @DeleteMapping("/{id}")
+    public ResponseEntity<Void> eliminarRecurso(
+        @PathVariable("id") Integer id,
+        @RequestParam("tipo") String tipo) {
+        
+        servicio.quitarRecurso(id, tipo);
+        return ResponseEntity.noContent().build();
+    }
+
+    @GetMapping("/buscar")
+    public Collection<Recurso> buscarRecursos(@RequestParam String criterio) {
+        return servicio.buscarRecursos(criterio);
+    }
```

Comment:
En REST, no usamos acciones (verbos) en las URI's de los recursos. Esas acciones se ven reflejadas en los métodos HTTP usados (GET, POST, PUT, etc). En este caso sugiero usar el siguiente endpoint: /api/recursos asi:

```suggestion
    @GetMapping
    public Collection<Recurso> buscarRecursos(@RequestParam String criterio) {
        return servicio.buscarRecursos(criterio);
    }
```